File: repos/firecrawl/apps/dot-net-sdk/Firecrawl/Exceptions/AuthenticationException.cs
namespace Firecrawl.Exceptions;

/// <summary>
/// Thrown when the API returns a 401 Unauthorized response.
/// </summary>
public class AuthenticationException : FirecrawlException
{
    public AuthenticationException(string message, string? errorCode = null, object? details = null)
        : base(message, 401, errorCode, details)
    {
    }
}

File: repos/firecrawl/apps/java-sdk/src/main/java/com/firecrawl/errors/AuthenticationException.java
package com.firecrawl.errors;

/**
 * Thrown when the API returns a 401 Unauthorized response.
 */
public class AuthenticationException extends FirecrawlException {

    public AuthenticationException(String message) {
        super(message, 401);
    }

    public AuthenticationException(String message, String errorCode, Object details) {
        super(message, 401, errorCode, details);
    }
}
